sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -349,10 +357,9 @@ class HDBSCAN(ClusterMixin, BaseEstimator):
     copy : bool, default=False
         If `copy=True` then any time an in-place modifications would be made
         that would overwrite data passed to :term:`fit`, a copy will first be
-        made, guaranteeing that the original data will be unchanged. Currently
-        this only makes a difference when passing in a dense precomputed
-        distance array (i.e. when `metric="precomputed"`) and using the
-        `"brute"` algorithm (see `algorithm` for details).
+        made, guaranteeing that the original data will be unchanged.
+        Currently, it only applies with `metric="precomputed"`, passing a dense
+        array or a sparse matrix of format CSR and algorithm used is `"brute"`.
```

Comment:
I think the current syntax is incorrect. Here is a rewording, what do you think?
```suggestion
        Currently, it only applies when `metric="precomputed"`, when passing
        a dense array or a CSR sparse matrix and when `algorithm="brute"`.
```